studiopress/atomic-blocks
# Migration

This PR adds migration features which install Genesis Blocks and redirect to the migration page in Genesis Blocks. 

### How to test
1. Ensure you see the admin notice to migrate. 
2. Click "migrate" to go to the migration page, click the "Install Genesis Blocks" button
3. Set your PHP version to something below PHP 7.1, and make sure you see the error telling you to upgrade.
4. Set your PHP version to something PHP 7.1 or newer, and make sure you see the "Install Genesis Blocks" button.
5. Make sure you do not have Genesis Blocks installed.
6. Disconnect from the internet.
7. Click the "Install Genesis Blocks" button. 
8. Make sure you get a failure notice that "Something went wrong" and the option to retry installing. 
9. Reconnect to the internet. 
10. Comment out the nonce in ab-migrate.js on line 10 to simulate a timed-out nonce.
11. Click the "Install Genesis Blocks" button. 
12. Make sure you get a failure notice that "You are not allowed to do that".
13. Undo what you did in step 6. 
14. Click the "Install Genesis Blocks" button. 
15. A spinner should show while you wait.
16. Upon successful installation, the spinner should hide, and you should see `Successfully installed and activated Genesis Blocks!`.
17. After 1 second, you should be automatically redirected to the [migrate page in Genesis Blocks](https://github.com/studiopress/genesis-blocks/pull/112). Note that at the moment this will show `Sorry, you are not allowed to access this page.` because that page does not yet exist in the version of Genesis Blocks that is pulled from the wp.org server. 

### Notes
This has been tested already by several team members in:
- https://github.com/studiopress/atomic-blocks/pull/406
- https://github.com/studiopress/atomic-blocks/pull/405
- https://github.com/studiopress/atomic-blocks/pull/404
